pallets/registrar/src/lib.rs

```diff
@@ -673,15 +685,16 @@ pub mod pallet {
             // Take the deposit immediately and give it to origin account
             if let Some(asset_info) = RegistrarDeposit::<T>::take(para_id) {
                 // Slash deposit from parachain creator
-                let (slashed_deposit, _not_slashed_deposit) =
-                    T::Currency::slash_reserved(&asset_info.creator, asset_info.deposit);
-                // _not_slashed_deposit can be greater than 0 if the account does not have enough reserved balance
-                // Should never happen but if it does, we just drop it
-                // TODO: is that the same as burning? Or as doing nothing? Either seems fine.
-
-                // Give deposit to origin account
                 // TODO: error handling
-                let _ = T::Currency::resolve_into_existing(&account, slashed_deposit);
+                let _ = T::Currency::transfer_on_hold(
```

Comment:
This is interesting, do we have a usecase for this? or a test? @tmpolaczyk do we have a test testing that if someone proves that you dont have ownership, the hold gets transferred to the prover?

Reply:
Yes, `can_deregister_with_empty_relay_state` and `can_deregister_pending_para` check that Bob is given Alice deposit